### DISCUSSION TITLE
Welcome to Data-Driven Machinery Fault Diagnosis Discussions!

### DISCUSSION THREAD

**Discussion by biswajitsahoo1111** [biswajitsahoo1111/cbm_codes_open#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members who find this repository useful. We hope that you:
  * Ask questions you’re wondering about.
  * Suggest further improvements.
  * Share your comments and criticisms.

  To get started, comment below elaborating your questions or comments.

**Comment by vinod151019** [comment]
1. Hey I didn't understand how did u resize the data into 32x32 can u send me the code.
2. I have one dataset can u resize it into your data so that i can use your code to get the accuracy using CNN please.

**Reply by biswajitsahoo1111** [reply]
Hi @vinod151019 ,

1. The dataset that I have used ([Case Western Reserve University Bearing dataset](https://csegroups.case.edu/bearingdatacenter/pages/welcome-case-western-reserve-university-bearing-data-center-website)) is a collection of time series data for different faults. In each time series (that I have used), there are approximately 480000 data points (actual number is slightly higher and different for each file). Each time series is then segmented into smaller segments of length 1024 such that there is no overlap between consecutive segments. After segmentation, each segment is resized into 32x32 so that it can be fed into a CNN layer. You can find all the details at this link: https://github.com/biswajitsahoo1111/cbm_codes_open/blob/master/notebooks/CWRU_time_domain_data_preprocessing.ipynb
2. If you have a time series like data, you can follow the same procedure given in the above notebook. By tweaking the code a little bit you would be able to resize your data. This is all about data preprocessing. Regarding getting good accuracy using CNN,  you have to fine tune you model by selecting appropriate hyperparameters. 

I hope this answers your questions. If you have any other questions or if you need further clarifications on some point,  please reply in this thread.
